tests/integration/utils.py

```diff
@@ -242,62 +255,73 @@ def assert_content_manifest_schema(response_data):
+        {<directory_name>: {files: {<file_name>: URL}, check_content: <boolean>}}
+    :param tmpdir: Testing function for providing temporary directory
+    :param files_to_ignore: The list of files which can be ignored in the assertion
     """
-    if expected_file_urls is None:
-        expected_file_urls = {}
-    assert os.path.exists(source_path) and os.path.isdir(
-        source_path
-    ), f"Invalid source_path: {source_path}"
-    files = []
-    # Go through all files in source_code_path and it's subdirectories
-    for root, _, source_files in os.walk(source_path):
-        for file_name in source_files:
-            # Get path to file in the project
-            absolute_file_path = os.path.join(root, file_name)
-            relative_file_path = os.path.relpath(absolute_file_path, start=source_path)
-            # Assert that content of source file is equal to expected
-            with open(absolute_file_path, "rb") as f:
-                if check_content:
-                    # Download expected file
-                    file_url = expected_file_urls[relative_file_path]
-                    expected_file = requests.get(file_url).content
-                    assert f.read() == expected_file
-                else:
-                    assert os.path.isfile(
-                        absolute_file_path
-                    ), f"File {absolute_file_path} does not exist"
-            files.append(relative_file_path)
-    # Assert that there are no missing or extra files
-    assert set(files) == set(list(expected_file_urls))
+    for test_source, archive_url in expected_files.items():
+        # A directory path to check
+        test_path = os.path.join(source_path, test_source)
+        # If there is no link to expected archive, the directory should be empty
+        if not archive_url:
+            assert (
+                len(os.listdir(test_path)) == 0
+            ), f"Directory: {test_path} not found or not empty as expected."
+        else:
+            # Dependencies instead of source code are saved as archives.
+            # Therefore we should extract it firstly
+            if test_source.startswith("deps"):
+                archive_path = test_path
+                data_path = os.path.join(tmpdir, "test_source")
+                extract_archive(archive_path, data_path)
+                dir_path = os.path.join(data_path, os.listdir(data_path)[0])
+            else:
+                dir_path = test_path
+            # A directory path with expected data
+            expected_data_dir = os.path.join(tmpdir, "expected_data")
+            # An archive path with expected data
+            expected_archive = os.path.join(tmpdir, "archive.tar.gz")
+            download_archive(archive_url, expected_archive)
+            extract_archive(expected_archive, expected_data_dir)
+            expected_directory = os.path.join(expected_data_dir, os.listdir(expected_data_dir)[0])
```

Comment:
can you change this variable name to be `expected_project_root_dir` or add a comment to say this is the root directory of the original package or dependency which will be compared to the cachito provided source?